src/cpp/src/icontinuous_batching.hpp

```diff
@@ -33,6 +33,8 @@ class ContinuousBatchingPipeline::IContinuousBatchingPipeline {
 
     PipelineMetrics m_pipeline_metrics;
 
+    std::string m_device;
```

Comment:
gtest fails. I'd guess the reason is that `CBForSDTest` inherits from `ContinuousBatchingPipeline` leaving `m_device` default initialized. Can you check if this inheritance really required?

Or maybe it's `PipelineTestInstance` inheriting from `ContinuousBatchingForSpeculativeDecodingImpl`